File: Tmp/Five/master4.patch.c
diff --git a/eg.c b/eg.c
index 6130a03..817df9d 100644
--- a/eg.c
+++ b/eg.c
@@ -10,7 +10,7 @@ int sum_name(char* name);
 
 int main(int argc, char** argv){
 
-  char name[26] = "abcde\0";
+  char name[26] = "defg\0";
   char greet[26];
   int value;
 
@@ -18,7 +18,7 @@ int main(int argc, char** argv){
   printf("%s", greet);
   printf("%c", '\n');
 
-  value = convert_silent(name);
+  value = convert(name);
 
   printf("%i", value);
   printf("%c", '\n');
@@ -36,19 +36,9 @@ int convert(char* name){
   char first_char;
   int converted_val;
 
-  first_char = name[0];
-
-  converted_val = 0;
-  if(first_char == 'a'){
-    printf("%c", 'A');
-    converted_val = 1;
-  }else if(first_char == 'b'){
-    printf("%c", 'B');
-    converted_val = 2;
-  }else if(first_char == 'c'){
-    printf("%c", 'C');
-    converted_val = 3;
-  }
+  converted_val = convert_silent(name);
+  //Characters start 'A', then 'B' etc
+  printf("%c", 'A'-1+converted_val);
   printf("%c", '\n');
 
   return converted_val;

File: Tmp/Five/master4.patch.f90
diff --git a/eg.f90 b/eg.f90
index 8497ba0..0790251 100644
--- a/eg.f90
+++ b/eg.f90
@@ -7,20 +7,9 @@ FUNCTION convert(name)
   CHARACTER(LEN=*), INTENT(IN) :: name
   CHARACTER :: first_char
 
-  first_char = name(1:1)
-
-  convert = 0
-  IF (first_char == 'a') THEN
-    PRINT*, 'A'
-    convert = 1
-  ELSE IF (first_char == 'b') THEN
-    PRINT*, 'B'
-    convert = 2
-  ELSE IF (first_char == 'c') THEN
-    PRINT*, 'C'
-    convert = 3
-  ENDIF
-
+  convert = convert_silent(name)
+  !Characters start at 64 for 'A', 65 for 'B' etc
+  PRINT*, char(convert+64)
 
 END FUNCTION
 
@@ -79,8 +68,8 @@ PROGRAM test
   greet = "Hello"
   PRINT*, greet
 
-  name = "abcde"
-  value = convert_silent(name)
+  name = "defg"
+  value = convert(name)
 
   PRINT*, value
 
